sample-plugin/src/main/java/com/github/jackchen/plugin/sample/instrumentation/SampleAsmClassVisitorFactory.kt

```diff
@@ -3,24 +3,52 @@ package com.github.jackchen.plugin.sample.instrumentation
 import com.android.build.api.instrumentation.AsmClassVisitorFactory
 import com.android.build.api.instrumentation.ClassContext
 import com.android.build.api.instrumentation.ClassData
-import com.android.build.api.instrumentation.InstrumentationContext
 import com.android.build.api.instrumentation.InstrumentationParameters
+import com.github.jackchen.android.sample.api.Extension
+import com.github.jackchen.android.sample.api.Register
+import com.github.jackchen.android.sample.api.TestCase
 import com.github.jackchen.plugin.sample.visitor.SampleClassVisitor
-import org.gradle.api.provider.Property
-import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.Internal
-import org.gradle.api.tasks.Optional
 import org.objectweb.asm.ClassVisitor
-import java.io.File
 
+/**
+ * @author JackChen
+ */
+abstract class SampleAsmClassVisitorFactory :
+  AsmClassVisitorFactory<InstrumentationParameters.None> {
 
-abstract class SampleAsmClassVisitorFactory : AsmClassVisitorFactory<InstrumentationParameters.None> {
+  companion object {
+    private val ANNOTATION_CLASS_NAMES = hashSetOf<String>(
+      Register::class.java.name,
+      TestCase::class.java.name,
+      Extension::class.java.name,
+    )
+  }
 
-    override fun createClassVisitor(classContext: ClassContext, nextClassVisitor: ClassVisitor): ClassVisitor {
-        return SampleClassVisitor(nextClassVisitor)
-    }
+  override fun createClassVisitor(
+    classContext: ClassContext,
+    nextClassVisitor: ClassVisitor
+  ): ClassVisitor {
+    return SampleClassVisitor(nextClassVisitor)
```

Comment:
Here you can add the class filter.

for example:

```
if(classContext.classData.className.endsWith("BuildConfig")|| ("R")){
    return nextClassVisitor.
}
return SampleClassVisitor
```

Reply:
There is no need to filter here, as it is already filtered out in the `isInstrumentable` method.